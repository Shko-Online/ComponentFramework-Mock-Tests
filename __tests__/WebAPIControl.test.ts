/*
    Copyright (c) 2022 Betim Beja and Shko Online LLC
    Licensed under the MIT license.
*/

import * as sinon from "sinon";

import {
  ComponentFrameworkMockGenerator,
  StringPropertyMock,
} from "@shko.online/componentframework-mock";
import { WebAPIControl } from "@powerapps-samples/web-api-control/WebAPIControl";
import {
  IInputs,
  IOutputs,
} from "@powerapps-samples/web-api-control/WebAPIControl/generated/ManifestTypes";
import resource from "@powerapps-samples/web-api-control/WebAPIControl/strings/WebAPIControl.1033.resx";

describe("WebAPIControl", () => {
  let mockGenerator: ComponentFrameworkMockGenerator<IInputs, IOutputs>;
  beforeEach(() => {
    window.alert = function (s: string) {
      console.info(s);
    };
    const container = document.createElement("div");

    mockGenerator = new ComponentFrameworkMockGenerator(
      WebAPIControl,
      {
        stringProperty: StringPropertyMock,
      },
      container
    );
    mockGenerator.SetControlResource(resource);
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.innerHTML = null;
  });
  it("Init should work", () => {
    mockGenerator.ExecuteInit();
    sinon.assert.calledOnce(mockGenerator.control.init);
    expect(document.body).toMatchSnapshot();
  });
  it("Update View should Work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    sinon.assert.calledOnce(mockGenerator.control.init);
    sinon.assert.calledOnce(mockGenerator.control.updateView);
    expect(document.body).toMatchSnapshot();
  });
  it("Button //100 work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();

    const button = mockGenerator.container.querySelector("#create_button_100");
    var evt = document.createEvent("Event");
    evt.initEvent("click", true, false);
    button.dispatchEvent(evt);

    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();
  });
  it("Button //200 work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();

    const button = mockGenerator.container.querySelector("#create_button_200");
    var evt = document.createEvent("Event");
    evt.initEvent("click", true, false);
    button.dispatchEvent(evt);

    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();
  });
  it("Button //300 work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();

    const button = mockGenerator.container.querySelector("#create_button_300");
    var evt = document.createEvent("Event");
    evt.initEvent("click", true, false);
    button.dispatchEvent(evt);

    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();
  });
  it("Button delete_button work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();

    const button = mockGenerator.container.querySelector("#delete_button");
    var evt = document.createEvent("Event");
    evt.initEvent("click", true, false);
    button.dispatchEvent(evt);

    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();
  });
  it("Button calculateAverage work", () => {
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();
    mockGenerator.context.webAPI.retrieveMultipleRecords.callsFake(
      (entity, options) => {
        return new Promise<ComponentFramework.WebApi.RetrieveMultipleResponse>(
          (resolve) => {
            resolve({
              entities: [
                {
                  average_val: 234,
                },
              ],
              nextLink: "string",
            });
          }
        );
      }
    );
    const buttons = mockGenerator.container.querySelectorAll("#odata_refresh");
    buttons.forEach((button) => {
      var evt = document.createEvent("Event");
      evt.initEvent("click", true, false);
      button.dispatchEvent(evt);

      mockGenerator.ExecuteUpdateView();
      expect(document.body).toMatchSnapshot();
    });
  });
  it("Destroy should work", () => {
    mockGenerator.control.destroy();
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();
  });
  it("getOuputs should work", () => {
    mockGenerator.control.getOutputs();
    mockGenerator.ExecuteInit();
    mockGenerator.ExecuteUpdateView();
    expect(document.body).toMatchSnapshot();
  });
});
