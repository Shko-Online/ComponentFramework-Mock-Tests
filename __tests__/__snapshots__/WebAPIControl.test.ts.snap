// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebAPIControl Button //100 work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button //100 work 2`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button //200 work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button //200 work 2`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button //300 work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button //300 work 2`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button calculateAverage work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button calculateAverage work 2`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button calculateAverage work 3`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button delete_button work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Button delete_button work 2`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Destroy should work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl Init should work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    />
  </div>
</body>
`;

exports[`WebAPIControl Update View should Work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;

exports[`WebAPIControl getOuputs should work 1`] = `
<body>
  <div>
    <div
      class="WebAPIControl_Container"
    >
      <div
        class="SampleControl_WebAPIControl_Header create_container_header"
      />
      <button
        buttonvalue="100"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_100"
      >
        Create record with annual revenue of 100
      </button>
      <button
        buttonvalue="200"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_200"
      >
        Create record with annual revenue of 200
      </button>
      <button
        buttonvalue="300"
        class="SampleControl_WebAPIControl_ButtonClass"
        id="create_button_300"
      >
        Create record with annual revenue of 300
      </button>
      <div
        class="SampleControl_WebAPIControl_Header delete_container_header"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="delete_button"
      >
        Select record to delete
      </button>
      <div
        class="SampleControl_WebAPIControl_Header fetchxml_status_container_header"
      />
      <div
        class="fetchxml_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Calculate average value of annual revenue
      </button>
      <div
        class="SampleControl_WebAPIControl_Header odata_status_container_header"
      />
      <div
        class="odata_status_container"
      />
      <button
        class="SampleControl_WebAPIControl_ButtonClass"
        id="odata_refresh"
      >
        Refresh record count
      </button>
      <div
        class="SampleControl_WebAPIControl_Header result_container_header"
      />
      <div
        class="result_container"
      >
        Web API sample custom control loaded
      </div>
    </div>
  </div>
</body>
`;
